cloudbuild.yaml

```diff
@@ -0,0 +1,24 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 1200s
+# this prevents errors if you don't use both _GIT_TAG and _PULL_BASE_REF,
+# or any new substitutions added in the future.
+options:
+  substitution_option: ALLOW_LOOSE
+steps:
+  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20210917-12df099d55'
+    entrypoint: bash
+    env:
+    - DOCKER_CLI_EXPERIMENTAL=enabled
+    - TAG=${_GIT_TAG}
+    - PULL_BASE_REF=${_PULL_BASE_REF}
+    args:
+    - cd admission-webhook && make release-staging
```

Comment:
I am not familiar with the promotion flow here.  This will push an image every time we merge a PR to main branch?  Does the test-infra job then promote the image to the non-staging repository? How do we promote only on a tagged version?

We should run at least unit tests and one integration test before promoting.

Reply:
The test infra job triggers the workflow which pushes changes to the staging images to `gcr.io/k8s-staging-gmsa-webhook`.
We can configure the triggers for the test infra job but most of the jobs do run on every PR merge into main.

We don't have a workflow for promoting images at the moment.

Reply:
My guess is we probably want the workflow for promoting image to trigger on tag pushes or pushes to release branches.

Reply:
> my guess is we probably want the workflow for promoting image to trigger on tag pushes or pushes to release branches.

yes this makes sense.

So the workflow here is:

- pr merges
- Prow post submit job triggers the cloudbuild which pushes to staging repository
- when we create tag image is promoted to official image (to be done in follow pr)

I guess we don't need the tests here then and looks like it common to simply build the image in the cloudbuild.yaml file but could we add a small section to the contributing readme so we have this documented?

Reply:
How about I add it to the release.md?

Reply:
I filed https://github.com/kubernetes-sigs/windows-gmsa/issues/60 and will document the release process in another PR.
The current release process is just a template and does not even reflect the process for releasing images to the old dockerhub account :(